packages/astro/src/assets/internal.ts

```diff
@@ -22,6 +23,30 @@ export function isESMImportedImage(src: ImageMetadata | string): src is ImageMet
 	return typeof src === 'object';
 }
 
+export function isRemoteImage(src: ImageMetadata | string): src is string {
+	return typeof src === 'string';
+}
+
+export function isRemoteAllowed(
+	src: string,
+	{
+		domains = [],
+		remotePatterns = [],
+	}: Partial<Pick<AstroConfig['image'], 'domains' | 'remotePatterns'>>
+): boolean | undefined {
+	let url: URL;
+	try {
+		url = new URL(src);
+	} catch (e) {
+		return undefined;
```

Comment:
We should **at least** comment and say what are the cases where this operation can fail.